[PATCH] Fix MMS not available when data connection is off.

Add key in CarrierConfig to support MMS when data connection is off or
the Sub is not the data Sub. If the flag is set to true, MMS will be
available no matter data connection is on or off.

diff --git a/telephony/java/android/telephony/CarrierConfigManager.java b/telephony/java/android/telephony/CarrierConfigManager.java
index e1deb98..eaa1411 100644
--- a/telephony/java/android/telephony/CarrierConfigManager.java
+++ b/telephony/java/android/telephony/CarrierConfigManager.java
@@ -523,6 +523,13 @@ public CarrierConfigManager() {
     public static final String KEY_ALLOW_NON_EMERGENCY_CALLS_IN_ECM_BOOL =
             "allow_non_emergency_calls_in_ecm_bool";
 
+    /**
+     * Flag indicating whether MMS data connection is allowed when mobile data is off.
+     * Default value is {@code false}, meaning that MMS is not available when mobile data is off.
+     * @hide
+     */
+    public static final String KEY_ALLOW_MMS_WHEN_MOBILE_DATA_OFF_BOOL = "allow_mms_when_mobile_data_off";
+
     /** The default value for every variable. */
     private final static PersistableBundle sDefaults;
 
@@ -630,6 +637,7 @@ public CarrierConfigManager() {
         sDefaults.putString(KEY_MMS_UA_PROF_URL_STRING, "");
         sDefaults.putString(KEY_MMS_USER_AGENT_STRING, "");
         sDefaults.putBoolean(KEY_ALLOW_NON_EMERGENCY_CALLS_IN_ECM_BOOL, true);
+        sDefaults.putBoolean(KEY_ALLOW_MMS_WHEN_MOBILE_DATA_OFF_BOOL, false);
     }
 
     /**
